modules_basic/compose.js

```diff
@@ -21,59 +22,64 @@ exports.needs = {
   file_input: 'first'
 }
 
-exports.gives = 'message_compose'
+exports.gives = {
+  'message_compose': true,
+  'mcss': true
+}
 
-function id (e) { return e }
+exports.create = function (api) {
+  return {
+    message_compose,
+    mcss: () => fs.readFileSync(__filename.replace(/js$/, 'mcss'), 'utf8')
+  }
 
-/*
-  opts can take
+  /*
+    opts can take
 
-    placeholder: string. placeholder text, defaults to "Write a message"
-    prepublish: function. called before publishing a message.
-    shrink: boolean. set to false, to make composer not shrink (or hide controls) when unfocused.
-*/
+      placeholder: string. placeholder text, defaults to "Write a message"
+      prepublish: function. called before publishing a message.
+      shrink: boolean. set to false, to make composer not shrink (or hide controls) when unfocused.
+  */
 
-exports.create = function (api) {
+  function message_compose (meta = {}, opts = {}, cb) {
+    if(!meta.type) throw new Error('message must have type')
 
-  return function (meta, opts, cb) {
     if('function' === typeof cb) {
-      if('function' === typeof opts)
+      if('function' === typeof opts) {
         opts = {prepublish: opts}
       }
-
-    if(!opts) opts = {}
+    }
     opts.prepublish = opts.prepublish || id
 
-    var accessories
-    meta = meta || {}
-    if(!meta.type) throw new Error('message must have type')
-    var ta = h('textarea', {
+    var actions
+
+    var textArea = h('textarea', {
       placeholder: opts.placeholder || 'Write a message',
-      style: {height: opts.shrink === false ? '200px' : ''}
+      className: opts.shrink === false ? '-expanded' : '-contracted'
```

Comment:
like the class names, but maybe we should move the class to the top "Compose" element?